### PR TITLE
GPC-198: Add acquirer queue metrics

### PR DESCRIPTION
**Pull request by oskarwilliams**

### CODE REVIEW

File: src/main/kotlin/uk/gov/gdx/datashare/repositories/AcquirerSubscriptionRepository.kt [hunk 1]
```diff
@@ -12,6 +12,8 @@ import java.util.*
 interface AcquirerSubscriptionRepository : CrudRepository<AcquirerSubscription, UUID> {
   fun findAllByWhenDeletedIsNull(): List<AcquirerSubscription>
 
+  fun findAllByQueueNameIsNotNullAndWhenDeletedIsNull(): List<AcquirerSubscription>
```
Comment (ethanmills) [inline src/main/kotlin/uk/gov/gdx/datashare/repositories/AcquirerSubscriptionRepository.kt:15]: Is there a test for this? 😁

File: src/main/kotlin/uk/gov/gdx/datashare/services/OutboundEventQueueService.kt [hunk 2]
```diff
@@ -302,4 +321,50 @@ class OutboundEventQueueService(
...
+      .metricDataQueries(dlqLengthQueries + ageOfOldestMessageQueries)
+      .startTime(Instant.now().minusSeconds(120))
+      .endTime(Instant.now())
+      .build()
+
+    val metricDataResults = cloudWatchClient.getMetricData(getMetricDataRequest).metricDataResults()
+    return queueNames.associateWith { queueName ->
+      QueueMetric(
+        metricDataResults.find { it.id() == queueName }?.values()?.firstOrNull()?.toInt() ?: 0,
+        metricDataResults.find { it.id() == dlqName(queueName) }?.values()?.firstOrNull()?.toInt() ?: 0,
```
Comment (ethanmills) [inline src/main/kotlin/uk/gov/gdx/datashare/services/OutboundEventQueueService.kt:343]: Do we want these to be zero on missing or something else? Surely missing is quite concerning and we want to be able to alarm on it?

File: src/main/kotlin/uk/gov/gdx/datashare/services/ScheduledJobService.kt [hunk 3]
```diff
@@ -16,17 +17,25 @@ class ScheduledJobService(
   private val acquirerEventRepository: AcquirerEventRepository,
   private val groApiService: GroApiService,
   private val supplierEventRepository: SupplierEventRepository,
-  meterRegistry: MeterRegistry,
+  private val meterRegistry: MeterRegistry,
+  private val outboundEventQueueService: OutboundEventQueueService,
 ) {
-  private val gauge =
+  private val unconsumedEventsGauge =
     meterRegistry.gauge("UnconsumedEvents", AtomicInteger(acquirerEventRepository.countByDeletedAtIsNull()))
```
Comment (ethanmills) [inline src/main/kotlin/uk/gov/gdx/datashare/services/ScheduledJobService.kt:24]: Not for now, but perhaps we ought to split this out like we have the queues? So per subscription

File: src/main/kotlin/uk/gov/gdx/datashare/repositories/AcquirerSubscriptionRepository.kt (hunk 1 above)
Comment (oskarwilliams) [reply src/main/kotlin/uk/gov/gdx/datashare/repositories/AcquirerSubscriptionRepository.kt:15]: YEs

File: src/main/kotlin/uk/gov/gdx/datashare/services/ScheduledJobService.kt (hunk 3 above)
Comment (oskarwilliams) [reply src/main/kotlin/uk/gov/gdx/datashare/services/ScheduledJobService.kt:30]: WIll do

Comment (oskarwilliams) [reply src/main/kotlin/uk/gov/gdx/datashare/services/ScheduledJobService.kt:30]: https://govukverify.atlassian.net/browse/GPC-397?atlOrigin=eyJpIjoiZTJlMmM4OTE5OGY3NGQzZjkzYjg2MDdhODM0ZjhkYjAiLCJwIjoiaiJ9

**Pull request merged**